scripts/features.sh

```diff
@@ -51,13 +51,13 @@ packages() {
 	shift
 
 	# We only allow variable names, parentheses and the operators: & | !
-	if [ "$(expr match "$cond" '.*[^A-Za-z0-9_()&|! ].*')" -gt 0 ]; then
+	if grep -q ".*[^A-Za-z0-9_()&|! ].*" <<< "$cond"; then
 		exit 1
 	fi
 
 	# Let will return false when the result of the passed expression is 0,
 	# so we always add 1. This way false is only returned for syntax errors.
-	local ret="$(env -i $vars bash --norc -ec "let _result_='1+($cond)'; echo -n \"\$_result_\"" 2>/dev/null)"
+	local ret="$(env -i "$vars" bash --norc -ec "let _result_='1+($cond)'; echo -n \"\$_result_\"" 2>/dev/null)"
```

Comment:
Again, we're relying on `$vars` being interpreted as multiple arguments, so the quotes break things.

Reply:
`${vars[@]}` should also work or not?

Reply:
As long as you leave out the quotes, yes, `${vars[@]}` and `$vars` seem to be equivalent. Not  sure if there would be a difference if *vars* were declared as some kind of array?

Reply:
hm, if I understood shellcheck docs right a quotes array should work:
https://github.com/koalaman/shellcheck/wiki/Sc2086#exceptions

Reply:
Yes, turning the string variable into an array sounds good.